pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiGetLastMessageIdTest.java

```diff
@@ -218,4 +225,40 @@ public void setTimeoutHandler(TimeoutHandler handler) {
         Assert.assertTrue(((MessageIdImpl)id[0]).getLedgerId() > 0);
         Assert.assertEquals( 2 * numberOfMessages -1, ((MessageIdImpl)id[0]).getEntryId());
     }
+
+    /**
+     * If a topic only have non-durable subscriptions or mark delete position of all the durable subscriptions
+     * are reached the LAC, all the ledgers except the current ledger will be deleted. Since the current ledger may not
+     * have any data, so the test is to ensure the get last message ID API can work in this case.
+     *
+     * In this case, the we should return the message ID (-1, -1).
+     */
+    @Test
+    public void testGetLastMessageIdWhenTopicWithoutData() throws Exception {
+        final String topic = "persistent://prop/ns-abc/testGetLastMessageIdWhenTopicWithoutData-" + UUID.randomUUID();
+        Producer<String> producer = pulsarClient.newProducer(Schema.STRING)
+                .topic(topic)
+                .create();
+        final int messages = 10;
+        for (int i = 0; i < messages; i++) {
+            producer.send("Message - " + i);
+        }
+        // To trigger the ledger rollover
+        admin.topics().unload(topic);
+        Topic topicRef = pulsar.getBrokerService().getTopicIfExists(topic).get().get();
+        ((PersistentTopic) topicRef).getManagedLedger().trimConsumedLedgersInBackground(new CompletableFuture<>());
+        Awaitility.await().untilAsserted(() -> {
+            PersistentTopicInternalStats stats = admin.topics().getInternalStats(topic);
+            Assert.assertEquals(stats.ledgers.size(), 1);
+        });
+        Reader<String> reader = pulsarClient.newReader(Schema.STRING)
+                .topic(topic)
+                .startMessageId(MessageId.earliest)
+                .create();
+        // This will call the get last message ID api.
+        boolean hasMessage = reader.hasMessageAvailable();
+        Assert.assertFalse(hasMessage);
+        MessageId messageId = admin.topics().getLastMessageId(topic);
```

Comment:
That's a little weird, I remembered that we are using (-1, -1, -1) to represent the earliest position. Why we don't throw a client exception to indicate that the consumer reaches the LAC position?https://github.com/apache/pulsar/blob/d2bc0f08aec15f739dcc3f47c381cb4f84254969/pulsar-client-api/src/main/java/org/apache/pulsar/client/api/MessageId.java#L81